### PR TITLE
feat: implementing columns block variant

### LINKED ISSUE #14: Column block
Column block that supports the following variations.
1. 
image

2. 
image

https://www.terrischeer.com.au/

### PR DESCRIPTION
**Pull request by anuj-adobe**
- Adding following two variants to columns block

 
**1. products**
image

**2. benefits**
image

[Sample Authoring Doc](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B9D8C0C52-77BD-4C72-AD5F-66BD38449C6B%7D&file=columns.docx&action=default&mobileredirect=true)

Fix #14 

Test URLs:
- Before: https://main--suncorp--aemsites.hlx.live/_draft/anuj/columns
- After: https://issue-14--suncorp--aemsites.hlx.live/_draft/anuj/columns

### CODE REVIEW

**Comment by JiangLong2019** [comment]
Add some comments on look & feel:
**Column products**
1. In mobile view, the title size looks different.
Original site:
image

EDS site:
image

2. Missing the gray line between columns.
image

3. Question - current site, the following buttons are included in the product section. Any idea to handle them?

image

4. Comparing with other icons, what's the difference that this one is included directly?
image



**Column benefits**
1. Mobile view shows different.
Original site:
image

EDS site:
image

2. Tablet view, icon and content are not aligned.
Original site:
image

EDS site:
image

File: blocks/columns/columns.css [hunk 1]
```diff
@@ -1,3 +1,27 @@
+.columns {
+  --products-icon-width: 3.75rem; /* 60px */
+  --products-icon-height: 3.4375rem; /* 55px */
+  --products-icon-font-size-mob: 3.75rem; /* 60px */
+  --products-icon-font-size-tab-desktop: 3.4375rem; /* 55px */
+  --products-name-height: 5rem; /* 80px */
+  --products-name-font-size-mob-desktop: 1.75rem; /* 28px */
+  --products-name-font-size-tablet: 1.25rem; /* 20px */
+  --products-name-font-color: black;
```
Comment (JiangLong2019) [inline blocks/columns/columns.css:9]: Nit: let's use the global property from styles.css

File: blocks/columns/columns.js [hunk 2]
```diff
@@ -1,9 +1,127 @@
...
+  col.querySelectorAll('p')?.forEach((para) => {
+    if (para.textContent.trim() === '') {
+      para.remove();
+    }
+  });
+}
+
 export default function decorate(block) {
   const cols = [...block.firstElementChild.children];
   block.classList.add(`columns-${cols.length}-cols`);
```
Comment (JiangLong2019) [inline blocks/columns/columns.js:118]: Seems the class is never used.

File: blocks/columns/columns.js [hunk 3]
```diff
@@ -1,9 +1,127 @@
...
+  const h3 = col.querySelector('h3');
+  if (titleIcon) {
+    productNameDiv.appendChild(titleIcon);
+  } else if (titlePicture) {
+    productNameDiv.appendChild(titlePicture);
+  }
+  if (h3) productNameDiv.appendChild(h3);
+
+  const productDescriptionDiv = document.createElement('div');
+  const descriptionPara = (titleIcon || titlePicture) ? col.querySelector('p:nth-of-type(2)') : col.querySelector('p:first-of-type');
```
Comment (JiangLong2019) [inline blocks/columns/columns.js:47]: Nit: about `p:nth-of-type(2)`, I would use the next element of h3 tag. The assumption is that the icon may not be authored but the title should be there. It depends on how it's authored, maybe not a problem at all.

File: blocks/columns/columns.js (hunk 2 above)
Comment (anuj-adobe) [reply blocks/columns/columns.js:118]: My understanding is to not change the default implementation of columns while implementing new variations.

File: blocks/columns/columns.js (hunk 3 above)
Comment (anuj-adobe) [reply blocks/columns/columns.js:47]: > The assumption is that the icon may not be authored but the title should be there.

I am handling this via conditional code -
1. if icon is present, description will be second <p>
2. if icon is not present, description will be first <p>

Am I missing any other case?

**Comment by anuj-adobe** [comment]
> Question - current site, the following buttons are included in the product section. Any idea to handle them?
image

> 
image

Probably we can create a new section for these buttons below the products section and add a style to make it visible only in mobile view. wdyt?

File: blocks/columns/columns.css (hunk 1 above)
Comment (anuj-adobe) [reply blocks/columns/columns.css]: done

**Comment by anuj-adobe** [comment]
> 4. Comparing with other icons, what's the difference that this one is included directly?

As discussed in standup, original site is using multiple source of icons -
1. font file 2. raw image 3. svg

I've added icons in EDS as present in original site.
Also, as per our discussion I've initiated a thread with the customer to understand if we can use every icon as svg
here: [Authoring Icons](https://teams.microsoft.com/l/message/19:81uxJmjaWTwuy-ZDVNEb_b-1LHQQBVKm31eRWOZ1px01@thread.tacv2/1724667200339?tenantId=36fb8e1d-a891-493f-96d0-3038e4e9291c&groupId=efba1680-2043-4e5c-8de1-bb7d7f7a78b8&parentMessageId=1724667200339&teamName=aem-suncorp&channelName=General&createdTime=1724667200339)

**Comment by anuj-adobe** [comment]
I've addressed the review comments via commit: https://github.com/aemsites/suncorp/pull/35/commits/a484db2ee672893c58284c7574eaccd969a78398 and also replied to few comments.

File: blocks/columns/columns.js (hunk 3 above)
Comment (JiangLong2019) [reply blocks/columns/columns.js:47]: How about this case? -  the icon is not present but the title and description are authored.
*Don't take so much time on this, we can add a description into Sidekick Library to tell customer which field is mandatory.

**Comment by JiangLong2019** [comment]
@anuj-adobe That's a good suggestion, please raise a ticket.

>Probably we can create a new section for these buttons below the products section and add a style to make it visible only in mobile view. wdyt?

**Review by JiangLong2019 — APPROVED** [review]

**Comment by JiangLong2019** [comment]
Approved. BTW, it's good to convert icon to SVG.

File: scripts/aem.js [hunk 4]
```diff
@@ -401,12 +401,15 @@ function decorateIcon(span, prefix = '', alt = '') {
   const iconName = Array.from(span.classList)
     .find((c) => c.startsWith('icon-'))
     .substring(5);
-  const img = document.createElement('img');
-  img.dataset.iconName = iconName;
-  img.src = `${window.hlx.codeBasePath}${prefix}/icons/${iconName}.svg`;
-  img.alt = alt;
-  img.loading = 'lazy';
-  span.append(img);
+
```
Comment (jacobrendina) [inline scripts/aem.js:404]: Given we're no longer going with the icon font approach, should these icon font changes (and the icon font files themselves) be removed from this PR before merging?

File: scripts/scripts.js [hunk 5]
```diff
@@ -41,6 +41,18 @@ async function loadFonts() {
   }
 }
 
+/**
+ * load icons.css and set a session storage flag
+ */
+async function loadIcons() {
```
Comment (jacobrendina) [inline scripts/scripts.js:47]: Same story here

**Comment by anuj-adobe** [comment]
@JiangLong2019 @jacobrendina 
I have removed the code, earlier added to support multiple icon sources in columns block.
Please have a final look and approve if it seems okay.

File: styles/styles.css [hunk 6]
```diff
@@ -340,4 +341,18 @@ main .section.highlight {
   footer .footer {
     max-width: 76.875rem;
   }
-}
\ No newline at end of file
+}
+
+main .section.benefits-background-plain {
```
Comment (jacobrendina) [inline styles/styles.css:346]: Would be good to merge the latest main branch into your working branch to ensure the tweaks to the section styling are all compatible

**Review by jacobrendina — COMMENTED** [review]
Just left one comment suggesting to merge the latest main into this working branch before pushing

**Comment by JiangLong2019** [comment]
@anuj-adobe Some findings.

1. Missing styling, are you planning to add the styling separately from current block?
image

2. The "Get a Quote" button doesn't work, I think the plan is to wait for Hero block to come in, and then you will merge the modal, right?

**Comment by anuj-adobe** [comment]
> @anuj-adobe The "Get a Quote" button doesn't work, I think the plan is to wait for Hero block to come in, and then you will merge the modal, right?

Yes exactly, I am not keeping modal implementation under this PR scope, we can merge it later. I can create an issue if that is not covered under existing list.

**Comment by anuj-adobe** [comment]
> @anuj-adobe That's a good suggestion, please raise a ticket.
> 
> > Probably we can create a new section for these buttons below the products section and add a style to make it visible only in mobile view. wdyt?

Created an issue for this -
https://github.com/aemsites/suncorp/issues/48

File: styles/styles.css (hunk 6 above)
Comment (anuj-adobe) [reply styles/styles.css:346]: Latest main is already merged.

**Review by jacobrendina — APPROVED** [review]

**Pull request merged**